waspc/data/Generator/templates/sdk/wasp/auth/forms/internal/common/LoginSignupForm.tsx

```diff
@@ -302,7 +302,7 @@ function AdditionalFormFields({
           disabled={isLoading}
         />
         {errors[field.name] && (
-          <FormError>{errors[field.name].message}</FormError>
+          <FormError>{errors[field.name]!.message}</FormError>
```

Comment:
I'm not sure why this is needed since we have the check `errors[field.name] && (`?

Reply:
Because, theoretically, the `errors` object could have a getter (a computed property) implemented like this:
```typescript
get someFieldName () {
    return Math.random() > 0.5 ? undefined : { message: "You're lucky this time" }
}
```
In more precise terms, object field access is a function call and can easily be impure.

To get around this, you can define a local `fieldError` variable above the JSX and then use that.

Reply:
Ok, fun fact, this is actually wrong. TypeScript will happily allow this error to happen. [Check it out](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBANgSwOYFsCGMC8MDeAoGQmJAU1ghBTIAsEwkYAKASlwKI4CcyBXTsGAFk0UagDpOaMABNKLGAD4YABjEBWGAH4YPGSQBmdEtJgAuGACIE0kihAX2hAL54XeBPqaJUaMRSqidEis+BygkLBQJAAeUObQnEFY8MjofpQ0QY4w4RRwJGJwIEiMUbFiUCAAqgAONSScAMJoECQszK547p6M3ugA2hb+mfQWALoh2bmRMXEwCUnYfWiDw4GjY1PgeQVFJWVQFdV1Dc2t7a5AA).


Looks like TS specifically doesn't like if you do it twice (access a field of one object to get a string for accessing the field of another object). But my suggestion above still works. See [here](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAogTnA9nAzlAvFAShAxsgEwB4Vg4BLAOwHMAaKAIhUQFsJgALK6hqAHygBXSgQgAzKhAIA+AFCzRuADYBDONHyVSUCAmQoAXLD2p5i1eqibtzNgDFyEJQSMBvKJRVsjpCjSgAvvIA9MFQAO7IANYosuRiUAAUukioANoMYCooKJFwBAwAugCUUK6yUJVWiFrAUMAQAB7APmTcGDomKBlZOXkFhbJBsqFQBIgQKJQA5HV5UXEJyV1pthAOTgQAdJ5sJWUVVdZ1Dc2tftQdKfqrrOuOzjteEIPDo-Oxx1ASm-CpVys1htHrsXoskj9nH9kKVylVqrV6k0WlBfO1MJCCNC4EMgA).